src/root.cpp

```diff
@@ -64,23 +88,44 @@ void Root::index(Cutelyst::Context *c)
 
 void Root::search(Context *c)
 {
-    auto result = m_db->findComponentsByString(c->request()->queryParam(QStringLiteral("q")));
+    /*const int appsPerPage = 100;
+    int offset;*/
+
+    QString searchText = c->req()->queryParam(QStringLiteral("q"));
+
+    auto result = m_db->search(c->request()->queryParam(QStringLiteral("q")));
     QVariantList apps;
     Q_FOREACH (const auto &desktop, result) {
         apps.append(componentToMap(c, desktop));
     }
     c->setStash(QStringLiteral("apps"), apps);
+
+    /*Pagination pagination(1000,
+                          appsPerPage,
+                          c->req()->queryParam(QStringLiteral("page"), QStringLiteral("1")).toInt());
+    offset = pagination.offset();
+    c->setStash(QStringLiteral("pagination"), pagination);
+    c->setStash(QStringLiteral("posts_count"), 100);
+    c->setStash(QStringLiteral("search"), searchText);*/
 }
 
 void Root::app(Context *c, const QString &id)
 {
-    auto desktop = m_db->componentById(id);
-    c->setStash(QStringLiteral("app"), componentToMap(c, desktop));
+    auto desktops = m_db->componentsById(id);
```

Comment:
And I also tend to avoid "auto" everywhere, I prefer to use only when the type is known (due right side) or for iterators